[ra2] ch5 re-word (#2138)
cntt-n/CNTT: doc/ref_arch/kubernetes/chapters/chapter05.md

- Image registries are monitored to automatically block or replace images with
known vulnerabilities, while also ensuring policies are used to gate what can be
deployed and who can deploy from the registry.
- Adopt a layered packaging model which supports separation of concerns during
image build.

The following functionalities are recommended for securing Kubernetes platforms:

- Image Certification (Scan for vulnerabilities) and Signing
- Role-base Access Control
- Secret Management
- How to overcome the lack of hard Kubernetes Cluster Multi-tenancy
  - Tenants without hard multi-tenancy requirements (multiple development teams
    in the same organisation) separated from each other by namespaces
  - For strict multi tenancy, a dedicated Kubernetes cluster per tenant should be used
- Integration with other security ecosystem like monitoring and alerting tools

## 5.2 Principles
The following are core principles to consider when securing cloud native applications:

- Deploy only secure applications and trusted codes
- Only deploy applications from validated and verified images
- Only deploy applications from trusted registries

> - Containers orchestration (Kubernetes) must be secure with administrative boundaries

Reviewer: ```suggestion
- Containers orchestration (Kubernetes) secured with administrative boundaries
```

management metadata on configuration, container application, and nodes
- Ensure all metadata is encrypted and network access runs over TLS connections

## 5.10  Enable Logging and Monitoring
Logging, monitoring, alerting and log aggregation are essential for Kubernetes.
Enable and monitor audit logs for anomalous or unwanted API calls, especially
any authorisation failure.

## 5.11  Run-Time Security
The following are recommended best practices for container run-time:
- Integrate run-time processes to Security Information and Event Monitoring (SIEM)
- Use container-aware run-time defence tools
- Ensure all running cloud native applications are from secure and verified images
- Cloud native applications are not run with root privileges
- Ensure sensitive workloads are properly segmented by namespaces or cluster to
mitigate the scope of compromise.

## 5.12  Secrets Management
It is recommended that the principle of least privilege is applied to secret
management in Kubernetes:
- Ensure that the cloud native applications can only read the secrets that these
applications need
- Have different set of secrets for different environments(like production,
development, and testing)

> Secret values protect sensitive data, it is recommended to protect them from
> unauthorised access. Ideally, they must be protected at rest and in transit.

Reviewer: ```suggestion
Secret values protect sensitive data, it is recommended to protect them from
unauthorised access. Ideally, by being protected at rest and in transit.
```

labels associated with images).

## 5.14  Orchestration & Container Manager
The Kubernetes orchestration manager also known as the control plane consist of
various components including a Kube-API server, an etcd storage, a
kube-controller-manager, a cloud-controller-manager, a kube-scheduler, and a DNS
server for Kubernetes services. The communication over these APIs needs to be
secured via different mechanisms like TLS encryption, API authentication via
LDAP etc. A control plane node in an unsecured boundary can lead to a potential
threat to the running workloads. It is recommended that a control plane node is
hardened in terms of security by disabling unused ports, prohibiting root access
etc.

They following are security recommendations for orchestration manager:

- Cluster management Network isolation can help protect the master node and
control where administrative commands can run. Use network isolation techniques,
configure RBAC on the cluster manager and configure node service accounts
following the principle of least privilege.
- Ensure that access control is applied to registries requiring unique credentials,
to limit who can control the build or add images.

> - Network access must run over TLS connections.

Reviewer: ```suggestion
- Network access runs over TLS connections.
```

using default namespaces.
- Enable Seccomp to ensure that the workloads have restricted actions available
within the container application.
- Limit discovery by restricting services and users that can access cluster
management metadata on configuration, containers and nodes

## 5.15  Security Perimeters
When applications or workloads run on Kubernetes, there are several layers which
come into picture that govern the security. Each of these layers needs to be
secured within their perimeters. The various layers that come into picture are:

- **Container Registry**: A container registry is a repository to manage container
**images. The access to container registry needs to be secured in order to
**prevent unauthorised access or image tampering.
- **Container Images**: Stored instance of a container that holds a set of
software needed to run an application. Before loading them to container
registry, they need to be secured by performing various checks like
vulnerability analysis, scans etc. These images should also be signed from
trusted sources.
- **Containers**: A lightweight and portable executable image that contains

> software and all of its dependencies. The containers need to be prevented from
> accessing the underlying OS like loading of kernel modules, mounting of
> directories of underlying OS etc and it must also be ensured that they don't
> run in privileged mode.
> - **Pods**: A Pod represents a set of running containers on your cluster.
> Kubernetes inherently offers pod security policies that define a set of
> conditions that a pod must run with in order to be accepted into the system.
> These policies help in ensuring the necessary checks for running the pods.
> - **Kubernetes Node**: A Kubernetes node in an unsecured boundary can lead to a

Reviewer: ```suggestion
software and all of its dependencies. The containers need to be prevented from
accessing the underlying OS like loading of kernel modules, mounting of
directories of underlying OS etc and ensuring that they don't
run in privileged mode.
- **Pods**: A Pod represents a set of running containers on your cluster.
Kubernetes inherently offers pod security policies that define a set of
conditions that a pod needs to run with in order to be accepted into the system.
These policies help in ensuring the necessary checks for running the pods.
- **Kubernetes Node**: A Kubernetes node in an unsecured boundary can lead to a
```
